nvm there is error if i manually edit cache to make it invalid. Maybe I should not print the usage.

```
Run "dotnet tool restore" to make the "t-rex" command available.
Usage: dotnet tool run [options] <COMMAND_NAME> [[--] <additional arguments>...]]

Arguments:
  <COMMAND_NAME>   The command name of the tool to run.

Options:
  -h, --help   Show command line help.
Additional Arguments:
  Arguments passed to the application that is being run.
```


Error from build machine binlog

```
Exec
    Assembly = Microsoft.Build.Tasks.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
    CommandLineArguments = dotnet tool run testSdkClean -- -noautoreporters -noRepoInference -dotnetPath F:\vsagent\92\s\artifacts\bin\Debug\dotnetLayout\dotnet -xml "F:\vsagent\92\s\artifacts\TestResults\Debug\Microsoft.NET.Clean.Tests_netcoreapp2.1_x64.xml" -html "F:\vsagent\92\s\artifacts\TestResults\Debug\Microsoft.NET.Clean.Tests_netcoreapp2.1_x64.html" -parallel none > F:\vsagent\92\s\artifacts\log\Debug\Microsoft.NET.Clean.Tests_netcoreapp2.1_x64.log
    Parameters
        Command = dotnet tool run testSdkClean -- -noautoreporters -noRepoInference -dotnetPath F:\vsagent\92\s\artifacts\bin\Debug\dotnetLayout\dotnet -xml "F:\vsagent\92\s\artifacts\TestResults\Debug\Microsoft.NET.Clean.Tests_netcoreapp2.1_x64.xml" -html "F:\vsagent\92\s\artifacts\TestResults\Debug\Microsoft.NET.Clean.Tests_netcoreapp2.1_x64.html" -parallel none > F:\vsagent\92\s\artifacts\log\Debug\Microsoft.NET.Clean.Tests_netcoreapp2.1_x64.log
        WorkingDirectory = F:\vsagent\92\s\artifacts\tmp\Debug\testSdkClean\
        IgnoreStandardErrorWarningFormat = True
    Run "dotnet tool restore" to make the "testSdkClean" command available. <-----------------------------------------here
    Errors
        F:\vsagent\92\s\src\Tests\Directory.Build.targets(70,5): error MSB3073: The command "dotnet tool run testSdkClean -- -noautoreporters -noRepoInference -dotnetPath F:\vsagent\92\s\artifacts\bin\Debug\dotnetLayout\dotnet -xml "F:\vsagent\92\s\artifacts\TestResults\Debug\Microsoft.NET.Clean.Tests_netcoreapp2.1_x64.xml" -html "F:\vsagent\92\s\artifacts\TestResults\Debug\Microsoft.NET.Clean.Tests_netcoreapp2.1_x64.html" -parallel none > F:\vsagent\92\s\artifacts\log\Debug\Microsoft.NET.Clean.Tests_netcoreapp2.1_x64.log" exited with code 1. [F:\vsagent\92\s\src\Tests\Microsoft.NET.Clean.Tests\Microsoft.NET.Clean.Tests.csproj]

```

--- a/src/Tests/Directory.Build.targets
+++ b/src/Tests/Directory.Build.targets
@@ -35,6 +35,7 @@
   <Target Name="TestAsTool" DependsOnTargets="Pack;LayoutSdk;_InnerGetTestsToRun">
     <PropertyGroup>
       <TestLocalToolFolder>$(ArtifactsTmpDir)$(ToolCommandName)\</TestLocalToolFolder>
+      <DOTNET_CLI_HOME>$(ArtifactsTmpDir)DOTNET_CLI_HOME\</DOTNET_CLI_HOME>
     </PropertyGroup>
 
     <RemoveDir Directories="$(TestLocalToolFolder)" />
@@ -47,13 +48,16 @@
     <RemoveDir Directories="$(NuGetPackageRoot)$(ToolCommandName)\$(PackageVersion)" />
 
     <Exec Command="dotnet new tool-manifest"
-      WorkingDirectory="$(TestLocalToolFolder)"/>
+      WorkingDirectory="$(TestLocalToolFolder)"
+      EnvironmentVariables="DOTNET_CLI_HOME=$(DOTNET_CLI_HOME)"/>
     
     <Exec Command="dotnet tool install --local $(ToolCommandName) --version $(PackageVersion) --add-source $(ArtifactsShippingPackagesDir)"
-      WorkingDirectory="$(TestLocalToolFolder)"/>
+      WorkingDirectory="$(TestLocalToolFolder)"
+      EnvironmentVariables="DOTNET_CLI_HOME=$(DOTNET_CLI_HOME)"/>
     
     <Exec Command="dotnet tool restore"
-      WorkingDirectory="$(TestLocalToolFolder)"/>
+      WorkingDirectory="$(TestLocalToolFolder)"
+      EnvironmentVariables="DOTNET_CLI_HOME=$(DOTNET_CLI_HOME)"/>
 
     <PropertyGroup>
       <ResultsXmlPath>@(TestToRun->'%(ResultsXmlPath)')</ResultsXmlPath>
@@ -69,6 +73,7 @@
 
     <Exec Command="dotnet tool run $(ToolCommandName) -- $(TestArgs)"
           WorkingDirectory="$(TestLocalToolFolder)"
+          EnvironmentVariables="DOTNET_CLI_HOME=$(DOTNET_CLI_HOME)"
           IgnoreStandardErrorWarningFormat="true" />
   </Target>
 

